pkg/runTest.py

```diff
@@ -27,16 +30,30 @@ def run(**kwargs):
     result_output = {}
     for test in data["tests"]:
         match = Matcher(
-            index=test["index"], level=logger_instance.level, ES_URL=ES_URL, verify_certs=False
+            index=test["index"],
+            level=logger_instance.level,
+            ES_URL=ES_URL,
+            verify_certs=False,
         )
-        result = process_test(
-            test, match, kwargs["output_path"], kwargs["uuid"], kwargs["baseline"]
+        result_dataframe = process_test(
+            test, match, kwargs["save_data_path"], kwargs["uuid"], kwargs["baseline"]
         )
-        if result is None:
+        if result_dataframe is None:
             return None
+        if kwargs["lookback"]:
```

Comment:
It appears we are not limiting our ES query to reduce the number of hits due to the lookback, but applying the lookback to the dataframe? Is this right? If so, wouldn't it be less complicated to add to the Matcher query a date range?

Reply:
Yes we are just applying to the dataframe, we could add it to the Matched query but this would need a change in fmatch.. would be adding a change in fmatch for better optimization.